spec/laa/fee_calculator/integration/errors_spec.rb

```diff
@@ -51,5 +51,144 @@
       include_examples 'has manyable errors', :units
       include_examples 'has manyable errors', :modifier_types
     end
+
+    context 'calculate' do
+      let(:fee_scheme) { client.fee_schemes(type: 'AGFS', case_date: '2018-01-01') }
+
+      subject(:calculate) do
+        fee_scheme.calculate(
+          scenario: scenario,
+          offence_class: offence_class,
+          advocate_type: advocate_type,
+          fee_type_code: fee_type_code,
+          day: days,
+          number_of_cases: number_of_cases,
+          number_of_defendants: number_of_defendants
+        )
+      end
+
+      let(:scenario) { 5 } # Appeal against convicition
+      let(:advocate_type) { 'JRALONE' }
+      let(:offence_class) { 'E' }
+      let(:fee_type_code) { 'AGFS_APPEAL_CON' }
+      let(:days) { 1 }
+      let(:number_of_cases) { 1 }
+      let(:number_of_defendants) { 1 }
+
+      context 'when not supplied with required params' do
+        subject(:calculate) do
+          fee_scheme.calculate(
+            scenario: scenario,
+            # fee_type_code: fee_type_code, # not supplied
+            offence_class: offence_class,
+            advocate_type: advocate_type,
+            day: days,
+            number_of_cases: number_of_cases,
+            number_of_defendants: number_of_defendants
+          )
+        end
+
+        it 'raise ResponseError' do
+          expect { calculate }.to raise_error(LAA::FeeCalculator::ResponseError, /is a required field/)
+        end
+      end
+
+      context 'when not supplied with required param values' do
+        context 'fee_type_code' do
+          let(:fee_type_code) { nil }
+
+          it 'raises ResponseError with message taken from response body' do
+            expect { calculate }.to raise_error(LAA::FeeCalculator::ResponseError, /fee_type_code must match/)
+          end
+        end
+
+        context 'scenario' do
+          let(:scenario) { nil }
+
+          it 'raises ResponseError with message taken from response body' do
+            expect { calculate }.to raise_error(LAA::FeeCalculator::ClientError, /Value.*Error/i)
```

Comment:
This might be more of a question related with what the server is returning, but why is the returned error different from the other ones where one of the required values is missing?

Reply:
have been talking to ian about this as these specs raised that exact issue. he has changed this in the API so will now rerun and rewrite specs to match.